### ISSUE TITLE
Upload using ynab API

### ISSUE THREAD

**Issue by ohthehugemanatee** [nuc/db-to-ynab#10]
It'd be a lot faster and more reliable!

https://api.youneedabudget.com/

**Comment by nuc** [comment]
I thought about that, but then we'd have to deal with de-duplication ourselves, no?

**Comment by ohthehugemanatee** [comment]
Nope, YNAB handles payee and transaction duplication through the API. Payees are automatically matched (and corresponding categories applied) by text. Transaction de-duplication is done by creating a unique transaction ID which combines the date, amount, and an index. See how I built that in the PR.
